dataclients/kubernetes/kube.go

```diff
@@ -564,7 +573,7 @@ func (c *Client) GetEndpointAddresses(ns, name string) []string {
 	if c.state == nil {
 		return nil
 	}
-	return c.state.GetEndpointsByName(ns, name, "TCP")
```

Comment:
This "TCP" looks suspicious.
We use GetEndpointAddresses only for redis updater and then trim "TCP://" prefix.
For endpoint slices now I see you need "TCP" too lookup port and "http" to format url.
Lets rename `backendProtocol` to `scheme` here and elsewhere.
I also think this change may break Redis updater because result will now have "http://" prefix, please check.

Reply:
The redis updater call path was also changed to trim http://. I think I want to re-do this in the future. "TCP" comes from the Kubernetes service (you can basically have their the values "TCP" or "UDP" or "STCP" and in our case it's always "TCP", because we do not support http3 nor quic as client).
On the other hand the backendProtocol (or scheme) is used to add the scheme to endpoints which is in general useful (and we might be able to get this from "appProtocol" in endpointslices) but for simplicity and as we support only http I think it's ok to have it hard coded right now.

Will rename backendProtocol->scheme

Reply:
done